### ISSUE TITLE
[Feature]: Only handle requests that do not have a response

### ISSUE THREAD

**Issue by djordon** [stacks-network/sbtc#587]
# Feature - Only handle requests that do not have a response

## 1. Description

When we reach out to the Emily API for deposit requests, we end up adding it to our list of deposits-to-watch-for each time. And every time there is a new bitcoin block we attempt to process "pending" deposits but our queries do not distinguish between deposits that do or do not have a response. This causes us to continually include deposit UTXOs in our bitcoin transactions.

### 1.1 Context & Purpose

Update our queries to exclude deposit requests that have a response transaction that has been confirmed ~~or submitted~~.

### 2.1 Acceptance Criteria:
   - [x] The signers' deposits-to-watch-for list never contains duplicates (this isn't critical, but nice to have). This was fixed in https://github.com/stacks-network/sbtc/pull/679.
   - [ ] The signers never construct a response transaction for a deposit request if they have responded already and it is confirmed on the canonical bitcoin blockchain.

## 3. Related Issues and Pull Requests (optional):

* This depends on ~~https://github.com/stacks-network/sbtc/issues/559~~ https://github.com/stacks-network/sbtc/issues/585.

**Comment by djordon** [comment]
This might be nothing to really worry about. We may end up doing more work than is necessary, but we cannot write duplicate records into the database so we're probably good.

**Comment by cylewitruk** [comment]
> confirmed or submitted

@djordon what is the definition of these two for this case?

**Comment by djordon** [comment]
> > confirmed or submitted
> 
> @djordon what is the definition of these two for this case?

You actually only care about it being confirmed, and confirmed means that the transaction has been included in a bitcoin block that is on the canonical bitcoin blockchain.

**Comment by cylewitruk** [comment]
@djordon so no confirmation wait time here? 1 block is enough? (I think we talked about this before, that the sBTC mints will be orphaned along with a bitcoin block and thus be void)

**Comment by djordon** [comment]
> @djordon so no confirmation wait time here? 1 block is enough? (I think we talked about this before, that the sBTC mints will be orphaned along with a bitcoin block and thus be void)

Yeah 1 block is enough. A reorg that orphans a bitcoin transaction will also reorg Stacks. On bitcoin the transaction re-enters the mempool and can be confirmed, but it can also happen that we do not confirm it if we remove it in an RBF. Although the contract call can be resubmitted, it will fail when executed once https://github.com/stacks-network/sbtc/pull/493 lands. This protects us in the case where a reorg leads to a deposit does not getting confirmed but the stacks transaction has been submitted.
